To: vim-dev@vim.org
Subject: Patch 7.3.006
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.006
Problem:    Can't build some multi-byte code with C89.
Solution:   Move code to after declarations. (Joachim Schmitz)
Files:	    src/mbyte.c, src/spell.c


*** ../vim-7.3.005/src/mbyte.c	2010-08-15 21:57:28.000000000 +0200
--- src/mbyte.c	2010-09-18 13:32:40.000000000 +0200
***************
*** 5167,5182 ****
      void
  xim_set_preedit()
  {
-     if (xic == NULL)
- 	return;
- 
-     xim_set_focus(TRUE);
- 
      XVaNestedList attr_list;
      XRectangle spot_area;
      XPoint over_spot;
      int line_space;
  
      if (!xim_has_focus)
      {
  	/* hide XIM cursor */
--- 5167,5182 ----
      void
  xim_set_preedit()
  {
      XVaNestedList attr_list;
      XRectangle spot_area;
      XPoint over_spot;
      int line_space;
  
+     if (xic == NULL)
+ 	return;
+ 
+     xim_set_focus(TRUE);
+ 
      if (!xim_has_focus)
      {
  	/* hide XIM cursor */
***************
*** 5554,5565 ****
      void
  xim_set_status_area()
  {
-     if (xic == NULL)
- 	return;
- 
      XVaNestedList preedit_list = 0, status_list = 0, list = 0;
      XRectangle pre_area, status_area;
  
      if (input_style & XIMStatusArea)
      {
  	if (input_style & XIMPreeditArea)
--- 5554,5565 ----
      void
  xim_set_status_area()
  {
      XVaNestedList preedit_list = 0, status_list = 0, list = 0;
      XRectangle pre_area, status_area;
  
+     if (xic == NULL)
+ 	return;
+ 
      if (input_style & XIMStatusArea)
      {
  	if (input_style & XIMPreeditArea)
*** ../vim-7.3.005/src/spell.c	2010-08-15 21:57:32.000000000 +0200
--- src/spell.c	2010-09-18 13:34:11.000000000 +0200
***************
*** 4679,4685 ****
      buf_T	*buf;
      char_u	fname[MAXPATHL];
  
!     /* Go through all buffers and handle 'spelllang'. */ //<VN>
      for (buf = firstbuf; buf != NULL; buf = buf->b_next)
  	ga_clear(&buf->b_s.b_langp);
  
--- 4679,4685 ----
      buf_T	*buf;
      char_u	fname[MAXPATHL];
  
!     /* Go through all buffers and handle 'spelllang'. <VN> */
      for (buf = firstbuf; buf != NULL; buf = buf->b_next)
  	ga_clear(&buf->b_s.b_langp);
  
*** ../vim-7.3.005/src/version.c	2010-09-14 12:47:30.000000000 +0200
--- src/version.c	2010-09-18 13:34:23.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     6,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
163. You go outside for the fresh air (at -30 degrees) but open the
     window first to hear new mail arrive.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
